src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -182,56 +188,48 @@ public void flush() {
    *
    * @return the number of items added to 'bundle'.
    */
-  public int drainNextBundleTo(Collection<? super E> outputCollection) {
+  public E takeBundle() {
     final Lock lock = this.lock;
     lock.lock();
     try {
-      Bundle outBundle = null;
+      E outBundle = null;
       if (closedBundles.size() > 0) {
         outBundle = closedBundles.remove(0);
```

Comment:
Hard to say without measurements. `ArrayDeque` theoretically works faster in steady-state, so maybe use that for now? We can always change later if measurements disagree.